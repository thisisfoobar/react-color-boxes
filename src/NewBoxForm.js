import { useState } from "react";

const NewBoxForm = ({ addBox }) => {
  const INITIAL_STATE = {
    boxWidth: "",
    boxHeight: "",
    boxColor: ""
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addBox({...formData});
    setFormData(INITIAL_STATE)
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="boxWidth">Width</label>
      <input
        id="boxWidth"
        type="number"
        min="1"
        max="200"
        name="boxWidth"
        placeholder="Box Width"
        value={formData.boxWidth}
        onChange={handleChange}
      />
      <label htmlFor="boxHeight">Height</label>
      <input
        id="boxHeight"
        type="number"
        min="1"
        max="200"
        name="boxHeight"
        placeholder="Box Height"
        value={formData.boxHeight}
        onChange={handleChange}
      />
      <label htmlFor="boxColor">Background Color</label>
      <input
        id="boxColor"
        type="text"
        name="boxColor"
        placeholder="Box Color"
        value={formData.boxColor}
        onChange={handleChange}
      />
      <button>Add new box</button>
    </form>
  )
};

export default NewBoxForm;