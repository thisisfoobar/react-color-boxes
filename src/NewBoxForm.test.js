import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it ("renders without crashing", function () {
  render(<NewBoxForm />)
})

it("matches the snapshot", function () {
  const { asFragment } = render(<NewBoxForm />)
  expect(asFragment()).toMatchSnapshot()
})

