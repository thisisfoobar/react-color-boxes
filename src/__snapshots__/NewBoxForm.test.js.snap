// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="boxWidth"
    >
      Width
    </label>
    <input
      id="boxWidth"
      max="200"
      min="1"
      name="boxWidth"
      placeholder="Box Width"
      type="number"
      value=""
    />
    <label
      for="boxHeight"
    >
      Height
    </label>
    <input
      id="boxHeight"
      max="200"
      min="1"
      name="boxHeight"
      placeholder="Box Height"
      type="number"
      value=""
    />
    <label
      for="boxColor"
    >
      Background Color
    </label>
    <input
      id="boxColor"
      name="boxColor"
      placeholder="Box Color"
      type="text"
      value=""
    />
    <button>
      Add new box
    </button>
  </form>
</DocumentFragment>
`;
